### PR TITLE
[v2] refactor MGW api

### PR DESCRIPTION
**Pull request by waynz0r**
| Q               | A
| --------------- | ---
| Bug fix?        |no
| New feature?    |no
| API breaks?     | yes
| Deprecations?   | no
| Related tickets | 
| License         | Apache 2.0

### What's in this PR?

Contains MGW api refactor to get rid of embedded structs.

### Why?

Unfortunately embedded structs cannot be fully supported with protobuf and causes more issues than it solves.

### CODE REVIEW

**Review by martonsereg — COMMENTED** [review]
could you please update the meshgateway sample to match the new structure?

File: api/v1alpha1/meshgateway.proto [hunk 1]
```diff
@@ -50,46 +50,55 @@ option go_package = "github.com/banzaicloud/istio-operator/v2/api/v1alpha1";
...
+
+    GroupVersionKind groupVersionKind = 1 [(gogoproto.nullable) = false];
+    NamespacedName objectKey = 2;
+    repeated Patch patches = 3 [(gogoproto.nullable) = false];
 }
 
 enum GatewayType {
-    Ingress = 0;
-    Egress = 1;
+    unspecified = 0;
```
Comment (Laci21) [inline api/v1alpha1/meshgateway.proto:99]: Why are these `unspecified`s needed in the Enums?

**Review by Laci21 — APPROVED** [review]
lgtm

File: api/v1alpha1/meshgateway.proto (hunk 1 above)
Comment (waynz0r) [reply api/v1alpha1/meshgateway.proto:99]: unspecified was introduced because the 0 value is not represented in marshalled json or yaml.

**Pull request merged**